src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int LIMIT = 10;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        Figure[] figures = new Figure[]
+                {new Circle(colorSupplier.getRandomColor(), random.nextInt(LIMIT)),
+                new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT)),
+                new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT),random.nextInt(LIMIT)),
+                new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(LIMIT), random.nextInt(LIMIT)),
+                new Square(colorSupplier.getRandomColor(), random.nextInt(LIMIT))};
+
+        int index = new Random().nextInt(5);
+        return figures[index];
```

Comment:
let's make 5 a constant https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#all-magic-numbers-in-your-code-should-be-constants